### ISSUE TITLE
semi-specified buffer size

### ISSUE THREAD

**Issue by bendichter** [catalystneuro/nwb-conversion-tools#326]
For a `GenericDataChunkIterator`, I'd like to be able to set a buffer shape to:

buffer_shape=(1, None)
buffer_gb=10

This would be useful when you know you only want to read one channel at a time, but don't know how many time points you want to read.

**Comment by CodyCBakerPhD** [comment]
On it - this might take a minute though, it always made the assumption that you specify only one or the other, not a partial of `buffer_shape`

**Comment by bendichter** [comment]
Yeah. I would have used this with the DJ project and I feel like this might be useful in other scenarios as well, but I think it's non-critical.
